pkg/oc/cli/cmd/image/mirror/mirror.go

```diff
@@ -0,0 +1,654 @@
+
+var schema2ManifestOnly = distribution.WithManifestMediaTypes([]string{
+	manifestlist.MediaTypeManifestList,
+	schema2.MediaTypeManifest,
+})
+
+func (o *pushOptions) Run() error {
+	tree := buildTargetTree(o.Mappings)
+
+	creds := importer.NewLocalCredentials()
+	ctx := apirequest.NewContext()
+
+	rt, err := rest.HTTPWrappersForConfig(&rest.Config{}, http.DefaultTransport)
+	if err != nil {
+		return err
+	}
+	insecureRT, err := rest.HTTPWrappersForConfig(&rest.Config{TLSClientConfig: rest.TLSClientConfig{Insecure: true}}, http.DefaultTransport)
+	if err != nil {
+		return err
+	}
+	srcClient := importer.NewContext(rt, insecureRT).WithCredentials(creds)
+	toContext := importer.NewContext(rt, insecureRT).WithActions("pull", "push")
+
+	var errs []error
+	for _, src := range tree {
+		srcRepo, err := srcClient.Repository(ctx, src.ref.DockerClientDefaults().RegistryURL(), src.ref.RepositoryName(), o.Insecure)
+		if err != nil {
+			errs = append(errs, retrieverError{err: fmt.Errorf("unable to connect to %s: %v", src.ref, err), src: src.ref})
+			continue
+		}
+
+		manifests, err := srcRepo.Manifests(ctx)
+		if err != nil {
+			errs = append(errs, retrieverError{src: src.ref, err: fmt.Errorf("unable to access source image %s manifests: %v", src.ref, err)})
+			continue
+		}
+
+		var tagErrs []retrieverError
+		var digestErrs []retrieverError
+
+		// convert source tags to digests
+		for srcTag, pushTargets := range src.tags {
+			desc, err := srcRepo.Tags(ctx).Get(ctx, srcTag)
+			if err != nil {
+				tagErrs = append(tagErrs, retrieverError{src: src.ref, err: fmt.Errorf("unable to retrieve source image %s by tag: %v", src.ref, err)})
+				continue
+			}
+			srcDigest := desc.Digest
+			glog.V(3).Infof("Resolved source image %s:%s to %s\n", src.ref, srcTag, srcDigest)
+			src.mergeIntoDigests(srcDigest, pushTargets)
+		}
+
+		canonicalFrom := srcRepo.Named()
+
+		for srcDigestString, pushTargets := range src.digests {
+			// load the manifest
+			srcDigest := digest.Digest(srcDigestString)
+			// var contentDigest digest.Digest / client.ReturnContentDigest(&contentDigest),
+			srcManifest, err := manifests.Get(ctx, digest.Digest(srcDigest), schema2ManifestOnly)
+			if err != nil {
+				digestErrs = append(digestErrs, retrieverError{src: src.ref, err: fmt.Errorf("unable to retrieve source image %s manifest: %v", src.ref, err)})
+				continue
+			}
+
+			// filter or load manifest list as appropriate
+			srcManifests, srcDigest, err := processManifestList(ctx, srcDigest, srcManifest, manifests, src.ref, o.includeDescriptor)
+			if err != nil {
+				digestErrs = append(digestErrs, retrieverError{src: src.ref, err: err})
+				continue
+			}
+			if len(srcManifests) == 0 {
+				fmt.Fprintf(o.ErrOut, "info: Filtered all images from %s, skipping\n", src.ref)
+				continue
+			}
+
+			for _, dst := range pushTargets {
+				// if we are going to be using cross repository mount, get a token that covers the src
```

Comment:
shouldn't this check if SkipMount is true?

Reply:
Probably, although it should always work regardless of cross mount.